libraries/AP_InertialSensor/AP_InertialSensor_Invensensev3.cpp

```diff
@@ -540,44 +540,50 @@ void AP_InertialSensor_Invensensev3::read_fifo()
 #else
     const uint8_t fifo_sample_size = INV3_SAMPLE_SIZE;
 #endif
-    if (!block_read(reg_counth, (uint8_t*)&n_samples, 2)) {
-        goto check_registers;
-    }
-
-    if (n_samples == 0) {
-        /* Not enough data in FIFO */
-        goto check_registers;
-    }
 
-    // adjust the periodic callback to be synchronous with the incoming data
-    // this means that we rarely run read_fifo() without updating the sensor data
-    dev->adjust_periodic_callback(periodic_handle, backend_period_us);
-
-    while (n_samples > 0) {
-        uint8_t n = MIN(n_samples, INV3_FIFO_BUFFER_LEN);
-        if (!block_read(reg_data, (uint8_t*)fifo_buffer, n * fifo_sample_size)) {
-            goto check_registers;
+    do {
```

Comment:
Do you know how much difference this makes? I ask because there is a bug in the v3 driver to do with DMA usage which we should fix - https://github.com/ArduPilot/ardupilot/pull/27841/commits/a56660e634eb7b8efe8b31ed4a5eca3c1eb7a0d7 - that said this also has the cost of two SPI transactions, so I am wondering if there is a way of doing both?